freqtrade/vendor/qtpylib/indicators.py

```diff
@@ -103,47 +104,53 @@ def session(df, start='17:00', end='16:00'):
 
     return df.copy()
 
-
 # ---------------------------------------------
 
+
 def heikinashi(bars):
     bars = bars.copy()
     bars['ha_close'] = (bars['open'] + bars['high'] +
                         bars['low'] + bars['close']) / 4
 
-    bars['ha_open'] = (bars['open'].shift(1) + bars['close'].shift(1)) / 2
-    bars.loc[:1, 'ha_open'] = bars['open'].values[0]
-    for x in range(2):
-        bars.loc[1:, 'ha_open'] = (
-            (bars['ha_open'].shift(1) + bars['ha_close'].shift(1)) / 2)[1:]
+    # ha open
+    idx = bars.index.name
+    bars.reset_index(inplace=True)
```

Comment:
I just pushed a fix for the tests - we simply need to reset the index, so we can save playing with the index here.

In addition - i would start range at 1 - and move the calculation of bar0 to before the loop.
It'll save a comparison per row - it's only a few ys in my 1m row test-setup - but then i have plenty of ram and my pc was bored, so it might make a difference on a small machine.

``` python
    bars.at[0, 'ha_open'] = (bars.at[0, 'open'] + bars.at[0, 'close']) / 2
    for i in range(1, len(bars)):
        bars.at[i, 'ha_open'] = (bars.at[i - 1, 'ha_open'] + bars.at[i - 1, 'ha_close']) / 2
```

Reply:
Okay, agree, done. This may have sense...